internal/wasm/engine.go

```diff
@@ -9,11 +9,9 @@ type Engine interface {
 
 // ModuleEngine implements function calls for a given module.
 type ModuleEngine interface {
-	// FunctionAddress returns the absolute address of the compiled function for index.
-	// The returned address is stored and used as a TableInstance.Table element.
-	//
-	// See https://www.w3.org/TR/2019/REC-wasm-core-1-20191205/#syntax-funcaddr
-	FunctionAddress(index Index) uintptr
+	// FunctionPointer returns the pointer the compiled function for index as interface{}.
+	// The returned interface is stored and used as a TableInstance.Table element.
+	FunctionPointer(index Index) interface{}
```

Comment:
```suggestion
	// Note: the result is a `uintptr`, but returned as `interface{}` to prevent errors like below running with `-race`:
	// 	fatal error: checkptr: pointer arithmetic result points to invalid allocation
	FunctionPointer(index Index) interface{}
```

Reply:
oh actually the result is *compiledFunction, not uintptr now

Reply:
whoops this is not correct either :D